packages/rocketchat-ui-master/client/main.js

```diff
@@ -154,6 +154,23 @@ Template.main.helpers({
 		const user = Meteor.user();
 		return user && user.requirePasswordChange === true;
 	},
+	require2faSetup() {
+		const user = Meteor.user();
+
+		// User is already using 2fa
+		if (user.services.totp !== undefined && user.services.totp.enabled) {
+			return false;
+		}
+
+		return user.roles.reduce((ret, role) => {
```

Comment:
this `reduce` could be changed to a single `findOne`:
```
RocketChat.models.Roles.findOne({ _id: { $in: user.roles }, mandatory2fa: true }})
```

Reply:
Changed.